src/Summoner/Source.hs

```diff
@@ -18,26 +18,34 @@ data Source
     = Url Text
     -- | File path to the local source file.
     | File FilePath
+    -- | Link to source source file.
+    | Link Text
     deriving (Show, Eq)
 
 sourceT :: Key -> BiToml Source
 sourceT nm = Toml.match (Toml._Text   >>> _Url)  (nm <> "url")
          <|> Toml.match (Toml._String >>> _File) (nm <> "file")
+         <|> Toml.match (Toml._Text >>> _Link) (nm <> "link")
   where
     _Url :: BiMap Text Source
-    _Url = Toml.invert $ Toml.prism (source Just (const Nothing)) Url
+    _Url = Toml.invert $ Toml.prism (source Just (const Nothing) (const Nothing)) Url
 
     _File :: BiMap FilePath Source
-    _File = Toml.invert $ Toml.prism (source (const Nothing) Just) File
+    _File = Toml.invert $ Toml.prism (source (const Nothing) Just (const Nothing)) File
 
-    source :: (Text -> c) -> (FilePath -> c) -> Source -> c
-    source f _ (Url x)  = f x
-    source _ f (File x) = f x
+    _Link :: BiMap Text Source
+    _Link = Toml.invert $ Toml.prism (source (const Nothing) (const Nothing) Just) Link
```

Comment:
I hope that with new `tomland` release there will be better way to write this code but should work for now!